### ISSUE TITLE
mosquitto_passwd -b is using username as password

### ISSUE THREAD

**Issue by jado** [eclipse/mosquitto#1949]
I am reinstalling my mqtt broker using 2.0.2, and the ansible role creates the passwd file using a loop with:

```
mosquitto_passwd -b /etc/mosquitto/passwd  user1  password1
```

None of the users could connect.
I confirmed on the command line that mosquitto_passwd is using the username as password.

I could connect using
```
username: user1
password: password1
```

**Comment by jado** [comment]
the error seems to be in mosquitto/apps/mosquitto_passwd/mosquitto_passwd.c

line 505
```
	}else if(batch_mode == true && idx+3 == argc){
		password_file_tmp = argv[idx];
		username = argv[idx+1];
		password_cmd = argv[idx+1];
```

should be

```
	}else if(batch_mode == true && idx+3 == argc){
		password_file_tmp = argv[idx];
		username = argv[idx+1];
		password_cmd = argv[idx+2];  <<<<--- change here
```

**Comment by ralight** [comment]
Thanks very much, I agree with your suggestion. That is now fixed and will be in 2.0.3.

**Comment by jado** [comment]
Great. Glad to help.
